osf/models/metadata.py

```diff
@@ -22,12 +38,39 @@ class Meta:
     def serializer(self):
         return serializer_registry[self.schema._id]
 
+    def validate_user_entered_metadata(self):
+        # TODO - consolodate this code which is from api.users.schemas.utils.from_json
+        here = os.path.split(os.path.abspath(__file__))[0]
+        with open(os.path.join(here, '../metadata/schemas/user_entered_datacite.json')) as f:
+            user_entered_schema = json.load(f)
+        # If validation fails, this will throw a validation error
+        return jsonschema.validate(self.metadata, user_entered_schema)
+
     def serialize(self):
         return self.serializer.serialize(self)
 
-    def validate(self, proposed_metadata):
-        return self.serializer.validate(self, proposed_metadata)
+    def validate(self):
+        return self.serializer.validate(self)
+
+    def update(self, proposed_metadata, user=None):
+        auth = Auth(user) if user else None
+        if auth and self.file.target.has_permission(user, osf_permissions.WRITE):
+            self.metadata = proposed_metadata
+            self.validate()
+            # TODO: manually call clean fields in update, or add this to save?
+            self.clean_fields()
```

Comment:
I know we roughly discussed in person, but is this what triggers the field level validator? If so, should `self.validate()` exist in the line above?